src/main/kotlin/com/autonomousapps/DependencyAnalysisPlugin.kt

```diff
@@ -457,6 +457,12 @@ class DependencyAnalysisPlugin : Plugin<Project> {
       output.set(outputPaths.pluginKaptAdvicePath)
     }
 
+    val adviceTransformTask = tasks.register<AdviceTransformTask>("adviceTransform$variantTaskName") {
```

Comment:
I would have preferred to declare it bedore `AdviceTask` but I couldn't. 
This task here will never be called and the only way I could find to have it configured lazily is to make `AdviceTask` being finalized by it. And this forces the task `AdviceTransformTask` to be declared before.
I write this comment out of curiosity, if you know a better way to do it @autonomousapps , I would be interested to learn about it.

Reply:
I had to do something similar here:
```kotlin
    // Configured below
    val failOrWarn = tasks.register<FailOrWarnTask>("failOrWarn")

    val adviceReport = tasks.register<AdviceAggregateReportTask>("adviceReport") {
      dependsOn(adviceReportsConf, advicePluginsConf)

      adviceReports = adviceReportsConf
      advicePluginReports = advicePluginsConf
      chatty.set(getExtension().chatty)

      projectReport.set(outputPaths.adviceAggregatePath)
      projectReportPretty.set(outputPaths.adviceAggregatePrettyPath)

      finalizedBy(failOrWarn)
    }

    failOrWarn.configure { ... }
```